### PR TITLE
[GEOT-6130] Extend parsing support for NetCDF units and make it configurable to support new units

### PR DESCRIPTION
**Pull request by aaime**

### CODE REVIEW

File: modules/library/metadata/src/main/java/org/geotools/util/GeoToolsUnitFormat.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
```
Comment (dromagnoli) [inline modules/library/metadata/src/main/java/org/geotools/util/GeoToolsUnitFormat.java]: Are the copyright headers 2004-2008 correct?

File: modules/library/metadata/src/main/java/org/geotools/util/GeoToolsUnitFormat.java [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+                Unit<?> baseUnit = tfmUnit.getParentUnit();
+                UnitConverter cvtr = tfmUnit.getConverter();
+                StringBuilder result = new StringBuilder();
+                // this is the one line that needs to be replaced
+                String baseUnitName = format(baseUnit);
+                String prefix = this.prefixFor(cvtr);
+                if (baseUnitName.indexOf(183) < 0
+                        && baseUnitName.indexOf(42) < 0
+                        && baseUnitName.indexOf(47) < 0) {
+                    result.append(baseUnitName);
```
Comment (dromagnoli) [inline modules/library/metadata/src/main/java/org/geotools/util/GeoToolsUnitFormat.java]: Curiosity of mine... not required to approve the PR :) ... what are these numbers?

File: modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/imageio/netcdf/NetCDFUnitFormat.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        SimpleUnitFormat format = AbstractNetCDFUnitFormat.getNewInstance();
+
+        // missing unit that cannot be expressed via config files
+        Unit bel = ONE.transform(new LogConverter(10));
+        format.alias(bel.divide(10), "dB");
+
+        // register non SI units as well
+        for (Unit u : NonSI.getInstance().getUnits()) {
+            if (u.getSymbol() != null && unknownSymbol(format, u.getSymbol())) {
+                System.out.println(u.getSymbol() + " -> " + u);
```
Comment (dromagnoli) [inline modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/imageio/netcdf/NetCDFUnitFormat.java]: System.out here... probably coming from tests during development.

File: modules/plugin/coverage-multidim/netcdf/src/test/java/RoundTripper.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import javax.measure.Unit;
+import org.geotools.imageio.netcdf.NetCDFUnitFormat;
+
+public class RoundTripper {
+
+    public static void main(String[] args) {
+        String unit = "µmol*m^-2*446.2";
```
Comment (dromagnoli) [inline modules/plugin/coverage-multidim/netcdf/src/test/java/RoundTripper.java]: Is that class needed?
Main + system.out :)

File: modules/library/metadata/src/main/java/org/geotools/util/GeoToolsUnitFormat.java (hunk 1 above)
Comment (aaime) [reply modules/library/metadata/src/main/java/org/geotools/util/GeoToolsUnitFormat.java]: It contains code copied from the UOM library, so I copied over the copyright header as-is

File: modules/library/metadata/src/main/java/org/geotools/util/GeoToolsUnitFormat.java (hunk 2 above)
Comment (aaime) [reply modules/library/metadata/src/main/java/org/geotools/util/GeoToolsUnitFormat.java]: I got it from a de-compiled version of the original library cause I had to change a single line of code to make parse/encode round trip actually work (they are not eager to fix it there, want use to upgrade to a different library that will be released in the future). Guess I can go and find the actual code, and see if I can use constants instead.

File: modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/imageio/netcdf/NetCDFUnitFormat.java (hunk 3 above)
Comment (aaime) [reply modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/imageio/netcdf/NetCDFUnitFormat.java]: Yep, will remove.

File: modules/plugin/coverage-multidim/netcdf/src/test/java/RoundTripper.java (hunk 4 above)
Comment (aaime) [reply modules/plugin/coverage-multidim/netcdf/src/test/java/RoundTripper.java]: Oh no, this was just a demo class that I used to report bugs to the UOM library, I'll remove

File: modules/library/metadata/src/main/java/org/geotools/util/GeoToolsUnitFormat.java (hunk 2 above)
Comment (dromagnoli) [reply modules/library/metadata/src/main/java/org/geotools/util/GeoToolsUnitFormat.java]: No need for that... It was simply curiosity of mine :)

**Pull request merged**